components/payloads/cu_ros_payloads/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+pub mod builtin;
+pub mod sensor_msgs;
+pub mod std_msgs;
+
+use serde::Serialize;
+
+/// ROS adaptation trait to convert payload data to ROS compatible message.
+/// The output type must match the structure of the related "msg" file.
+/// The namespace relates to the ROS namespace (such as "std_msgs")
+/// and the type name is the same as the message filename.
+pub trait RosMsgAdapter {
+    type Output: Serialize;
+
+    fn namespace() -> &'static str;
+    fn type_name() -> &'static str;
+    fn convert(&self) -> Self::Output;
+}
```

Comment:
I am not sure about forcing static here, if you parametrize the type I think the implementor/users of that trait can always use static for it:

```Rust
pub trait RosMsgAdapter<'a> {
    type Output: Serialize;

    fn namespace() -> &'a str;
    fn type_name() -> &'a str;
    fn convert(&self) -> Self::Output;
}
``` 

```Rust
impl<'a> RosMsgAdapter<'a> for MyMsg {
    type Output = MyRosCompatibleMsg;

    fn namespace() -> &'a str {
        "std_msgs"
    }

    fn type_name() -> &'a str {
        "String"
    }

    fn convert(&self) -> Self::Output {
        MyRosCompatibleMsg { ... }
    }
}
```

Reply:
Let me known if it sounds better now.

Reply:
I was thinking about the convert method ... this is just a not very idiomatic way of a From/Into, you can force the trait with a + 

```
pub trait RosMsgAdapter<'a>: Into<Self::Output> + Sized
where
    Self::Output: Serialize + 'a
{
    fn namespace() -> &'a str;
    fn type_name() -> &'a str {
        ros_type_name!(Self::Output)
    }
}
```
^ that feels more like the signature I would expect